src/core/shards/inlined.cpp

```diff
@@ -0,0 +1,154 @@
+  if (name == "Const") {
+    shard->inlineShardId = InlineShard::CoreConst;
+  } else if (name == "Pass") {
+    shard->inlineShardId = InlineShard::NoopShard;
+  } else if (name == "OnCleanup") {
+    shard->inlineShardId = InlineShard::NoopShard;
+  } else if (name == "Comment") {
+    shard->inlineShardId = InlineShard::NoopShard;
+  } else if (name == "Input") {
+    shard->inlineShardId = InlineShard::CoreInput;
+  } else if (name == "Pause") {
+    shard->inlineShardId = InlineShard::CoreSleep;
+  } else if (name == "ForRange") {
+    shard->inlineShardId = InlineShard::CoreForRange;
+  } else if (name == "Repeat") {
+    shard->inlineShardId = InlineShard::CoreRepeat;
+  } else if (name == "Once") {
+    shard->inlineShardId = InlineShard::CoreOnce;
+  } else if (name == "Set") {
+    shard->inlineShardId = InlineShard::CoreSet;
+  } else if (name == "Update") {
+    shard->inlineShardId = InlineShard::CoreUpdate;
+  } else if (name == "Swap") {
+    shard->inlineShardId = InlineShard::CoreSwap;
+  } else if (name == "Push") {
+    shard->inlineShardId = InlineShard::CorePush;
+  } else if (name == "Is") {
+    shard->inlineShardId = InlineShard::CoreIs;
+  } else if (name == "IsNot") {
+    shard->inlineShardId = InlineShard::CoreIsNot;
+  } else if (name == "IsMore") {
+    shard->inlineShardId = InlineShard::CoreIsMore;
+  } else if (name == "IsLess") {
+    shard->inlineShardId = InlineShard::CoreIsLess;
+  } else if (name == "IsMoreEqual") {
+    shard->inlineShardId = InlineShard::CoreIsMoreEqual;
+  } else if (name == "IsLessEqual") {
+    shard->inlineShardId = InlineShard::CoreIsLessEqual;
+  } else if (name == "And") {
+    shard->inlineShardId = InlineShard::CoreAnd;
+  } else if (name == "Or") {
+    shard->inlineShardId = InlineShard::CoreOr;
+  } else if (name == "Not") {
+    shard->inlineShardId = InlineShard::CoreNot;
+  } else if (name == "Math.Add") {
+    shard->inlineShardId = InlineShard::MathAdd;
+  } else if (name == "Math.Subtract") {
+    shard->inlineShardId = InlineShard::MathSubtract;
+  } else if (name == "Math.Multiply") {
+    shard->inlineShardId = InlineShard::MathMultiply;
+  } else if (name == "Math.Divide") {
+    shard->inlineShardId = InlineShard::MathDivide;
+  } else if (name == "Math.Xor") {
+    shard->inlineShardId = InlineShard::MathXor;
+  } else if (name == "Math.And") {
+    shard->inlineShardId = InlineShard::MathAnd;
+  } else if (name == "Math.Or") {
+    shard->inlineShardId = InlineShard::MathOr;
+  } else if (name == "Math.Mod") {
+    shard->inlineShardId = InlineShard::MathMod;
+  } else if (name == "Math.LShift") {
+    shard->inlineShardId = InlineShard::MathLShift;
+  } else if (name == "Math.RShift") {
+    shard->inlineShardId = InlineShard::MathRShift;
+  }
+  // Unary math is dealt inside math.hpp compose
+}
+
+SHARDS_CONDITIONAL_INLINE FLATTEN bool activateShardInline(Shard *blk, SHContext *context, const SHVar &input, SHVar &output) {
```

Comment:
well this won't work, cos it's in a .cpp file

Reply:
So when it's enabled, it'll compile [this file](https://github.com/fragcolor-xyz/shards/blob/478a33940040c9d06dfc3febc49e839c319ee984/src/core/runtime_inlined.cpp)
which groups everything into a single compilation unit, also checked the assembly to make sure it still does.

Reply:
ah ok good job